### ISSUE TITLE
Segmentation fault when installed through pyenv (brew's python OK, system python OK)

### ISSUE THREAD

**Issue by plmrn** [Kozea/WeasyPrint#2068]
Hi,

I'm having an issue when trying to install weasyprint through pip on a pyenv virtualenv: I run into a segmentation fault everytime I try to render a PDF.

I installed weasyprint through homebrew and this one works well, but weasyprint installed through pip creates a segmentation fault (whichever version of python I use on the virtualenv).

Is there any way to use a virtual environment but link weasyprint to the one that's actually working (in homebrew)? I'm not the most skilled when it comes to virtual environments and so on so I might be missing something obvious, in which case sorry for the time wasted in advance. But I did spend several hours looking into this online and finding similar but not quite the same other cases.

I need to use a virtual environment because I have several python projects living on my laptop.

I'm running Mac OS 14.3 on a 2021 MacBookPro with a M1 Pro chip.

Obviously let me know what extra info/steps can help to identify the issue.

Thanks a lot (it's probably lack of knowledge but it has been driving me crazy otherwise I wouldn't have resorted to posting here)

**Comment by liZe** [comment]
Hi,

(Disclaimer: we don’t have a Mac at home to test, so we won’t be able to help you much about this. That’s why we only support Homebrew’s version of WeasyPrint, Homebrew devs do an incredible work to make everything work well and take care of problems like this one!)

What you can try (if you didn’t already try) is to create a virtual environment using Homebrew’s Python. `python3` is supposed to lead to Homebrew’s version of Python (you can check that with `which python3`), so something like:
```sh
$ brew install weasyprint
$ python3 -m venv venv
$ venv/bin/pip install weasyprint
$ venv/bin/weasyprint --info
```
should work.

If it doesn’t work, you can create a virtual environment that has access to system site packages:

```sh
$ brew install weasyprint
$ python3 -m venv venv
$ venv/bin/python3 -m weasyprint --info
```

Let’s hope it works for you! 🤞🏽

**Comment by plmrn** [comment]
Thank you SO MUCH!
I'm very sorry as I realize it's not weasyprint-related, I just got confused between system, homebrew, and my virtual envs that I used to create with `pyenv virtualenv`.
Creating a virtual env with homebrew's python@3.12 (I needed 3.12 to be aligned with my heroku config) worked perfectly.

At least now I have a better grasp of python and virtual envs on MacOS.

Thanks again!
Best
Paul
